src/TwigComponent/src/Attribute/TwigComponent.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent\Attribute;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class TwigComponent
+{
+    private string $name;
+    private ?string $template;
+
+    public function __construct(string $name, ?string $template = null)
+    {
+        $this->name = $name;
+        $this->template = $template;
+    }
+
+    final public function getName(): string
+    {
+        return $this->name;
+    }
+
+    final public function getTemplate(): string
+    {
+        return $this->template ?? "components/{$this->name}.html.twig";
+    }
+
+    /**
+     * @internal
+     */
+    final public static function forClass(string $class): ?static
```

Comment:
The `static` return seems to be what's causing the `coding-style-php` job to fail.

Reply:
which `static`? The first or the second?

Reply:
Sorry, the return type. It's new in PHP8 and coding-style-php runs on PHP7.4 (and an older version of php-cs-fixer).